### PR TITLE
Add ECS EC2 Launch Type Daemon Deployment test workflow

### PR DESCRIPTION
**Pull request by ChenaLee**
# Description of the issue
Need to add ECS tests for metrics

# Description of changes
Added ECS EC2 Launch Type Daemon Deployment test workflow

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Tested on local fork. Need the test package to merge first: https://github.com/aws/amazon-cloudwatch-agent-test/pull/26

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make linter`

### CODE REVIEW

File: .github/workflows/integrationTest.yml [hunk 1]
```diff
@@ -675,6 +680,68 @@ jobs:
       - name: Terraform destroy
         run: terraform destroy --auto-approve
 
+  ECSEC2LaunchDaemonIntegrationTest:
```
Comment (SaxyPandaBear) [inline .github/workflows/integrationTest.yml:683]: Not a problem to solve here, and not even sure if it's technically feasible, but it would be nice if we could separate out these integ test jobs because this YAML file is really long and hard to follow now.

**Review by SaxyPandaBear — APPROVED** [review]

**Comment by williazz** [comment]
nitpicks, not a big deal:

- why are secrets remapped from `secrets` context to `env`? They were already accessible
- and isn't it best practice to keep commented code on a separate dev branch?

File: .github/workflows/integrationTest.yml (hunk 1 above)
Comment (khanhntd) [reply .github/workflows/integrationTest.yml:683]: Yes. We can separate the workflow and use dispatch to trigger appropriate workflow https://github.com/peter-evans/repository-dispatch/releases

File: .github/workflows/integrationTest.yml [hunk 2]
```diff
@@ -144,13 +147,15 @@ jobs:
           echo "::set-output name=ec2_gpu_matrix::$(echo $(cat generator/resources/ec2_gpu_complete_test_matrix.json))"
```
Comment (williazz) [inline .github/workflows/integrationTest.yml:147]: **nitpick**: `::set-output` is deprecated. [docs](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)

File: .github/workflows/integrationTest.yml (hunk 1 above)
Comment (SaxyPandaBear) [reply .github/workflows/integrationTest.yml:683]: We are not pulling in a dependency for an unverified GitHub action.

**Comment by SaxyPandaBear** [comment]
@williazz where do you see `secrets` -> `env` and what commented code are you referring to?

**Comment by williazz** [comment]
@SaxyPandaBear it's not in this PR, but it's in the workflow file. I was just asking in general

secrets are remapped to env on lines 5-17, and commented out code + todos are at the end

File: .github/workflows/integrationTest.yml (hunk 1 above)
Comment (khanhntd) [reply .github/workflows/integrationTest.yml:683]: Maybe I'm wrong but I think its has MIT License and also its only serve for the purpose of Github Action, not any Security Concern.

File: .github/workflows/integrationTest.yml [hunk 3]
```diff
@@ -690,6 +757,7 @@ jobs:
       - uses: actions/checkout@v2
         with:
           repository: ${{env.CWA_GITHUB_TEST_REPO_NAME}}
+          ref: ${{env.CWA_GITHUB_TEST_REPO_BRANCH}}
```
[the remark this answers is not in the document]
Comment (ChenaLee) [reply .github/workflows/integrationTest.yml:760]: which ones? I thought I added to all that uses the test package

File: .github/workflows/integrationTest.yml (hunk 2 above)
Comment (ChenaLee) [reply .github/workflows/integrationTest.yml:147]: yeah we've been having this warning forever. probably have to make a sim to address all warnings. There's another one complaining about node 12.x or something

**Comment by ChenaLee** [comment]
@billy is it the best practice? For.. readability? Also you can comment on specific line(s) in files changed tab

File: .github/workflows/integrationTest.yml (hunk 2 above)
Comment (williazz) [reply .github/workflows/integrationTest.yml:147]: the node 12.x thing might be fixed by upgrading checkout to v3

Comment (SaxyPandaBear) [reply .github/workflows/integrationTest.yml:147]: Not worth blocking on this PR.

File: .github/workflows/integrationTest.yml [hunk 4]
```diff
@@ -675,6 +680,68 @@ jobs:
...
+              -var="ec2_instance_type=${{ matrix.arrays.instanceType }}" \
+              -var="ami=${{ matrix.arrays.ami }}" ; then 
+              terraform destroy -auto-approve
+            else
+              terraform destroy -auto-approve && exit 1
+            fi
+
+      - name: Terraform destroy
+        if: ${{ cancelled() && steps.ecs-ec2-launch-integration-test.outputs.cache-hit != 'true' }}
+        uses: nick-invision/retry@v2
```
Comment (SaxyPandaBear) [inline .github/workflows/integrationTest.yml:738]: Wasn't there a PR that uses a different GH action? A lot of PRs flew in while I was out but I remember seeing something like that. Does this need to be changed? Or is this the correct, new action?

File: .github/workflows/integrationTest.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (SaxyPandaBear) [reply .github/workflows/integrationTest.yml:760]: Where are you suggesting this change?

**Review by khanhntd — APPROVED** [review]
